packages/react-redux-subspace/src/components/SubspaceProvider.js

```diff
@@ -25,22 +58,40 @@ class SubspaceProvider extends React.PureComponent {
     }
 }
 
+SubspaceLegacyContextOutput.childContextTypes = {
+    store: PropTypes.object
+}
+
+class SubspaceProvider extends React.Component {
+    makeSubspace = memoizeOne((mapState, namespace, subspaceDecorator, store) => {
+        const createSubspace = subspaceDecorator || subspace(mapState, namespace)
+        return createSubspace(store)
+    })
+
+    render() {
+        const { mapState, namespace, subspaceDecorator, store, children } = this.props
+        const subspace = this.makeSubspace(mapState, namespace, subspaceDecorator, store)
+        return (
+            <Provider key={subspace.namespace} store={subspace}>
+                <SubspaceLegacyContextOutput subspace={subspace}>
+                    {Children.only(children)}
+                </SubspaceLegacyContextOutput>
+            </Provider>
+        )
+    }
+}
+
 SubspaceProvider.propTypes = {
     children: PropTypes.element.isRequired,
+    store: PropTypes.object.isRequired,
     mapState: PropTypes.oneOfType([
         PropTypes.func,
         PropTypes.string,
     ]),
     namespace: PropTypes.string,
-    subspaceDecorator: PropTypes.func,
-}
-
-SubspaceProvider.contextTypes = {
-    store: PropTypes.object.isRequired
+    subspaceDecorator: PropTypes.func
 }
 
-SubspaceProvider.childContextTypes = {
-    store: PropTypes.object
-}
+const storeAccess = compose(contextApi, legacyContext)
```

Comment:
2 way to access the store, both pass the store as a prop to the underlying `SubspaceProvider` only if they find it in their respective methods.  Technically with this change, a store can be passed directly as a prop if there is no `Provider` (or `SubspaceProvider`) higher in the tree.

Reply:
Providing your own store could be useful for testing?